### PR TITLE
Reset nvim_lsp_fetching to true before RunAsync

### LINKED ISSUE #438: Potential bug in `Run()` of `nvim_lsp.vim`
**Describe the bug**
From second time run `Vista finder nvim_lsp` in a new file will give the stale output which is the result of last file.

**Environment:**
- OS: Ubuntu 18.04
- Vim/Neovim version: NVIM v0.8.0-dev+105-ge5f6f2096
- This plugin version: f9c8c8ad4135346a945e1ebfb00b723976d92dfc
- I'm using some LSP client:
    - Related Vim LSP client: nvim_lsp

**Steps to reproduce given the above info**

1. open a new code file let's say `a.rs`, and run `Vista finder nvim_lsp`, this will give the right output.
2. open another code file let's say `b.rs`, and run `Vista finder nvim_lsp`, this will give the output same with `a.rs`, and it's not the correct output of `b.rs`.
3. keep on `b.rs` and run another time `Vista finder nvim_lsp`, this time will give the correct output of `b.rs`.
4. after first file, every new file will give the stale output of first run `Vista finder nvim_lsp` and second run is correct.

**Expected behavior**
Every run of `Vista finder nvim_lsp` will give the right output.

**Screenshot or gif** (if possible)
Run on https://github.com/liuchengxu/vim-clap
1. open `crates/matcher/src/substring.rs` and run `Vista finder nvim_lsp`
image
2. open `crates/matcher/src/mod.rs` and run `Vista finder nvim_lsp`
image
3. keep on `crates/matcher/src/mod.rs` and run another time `Vista finder nvim_lsp`
image

**Potential root cause**
1. Every time a new file is opened and run `Vista finder nvim_lsp`, cache will be empty so a `vista#executive#nvim_lsp#Run` and then `s:Run` will be call. 
2. I think it's designed to call `s:RunAsync()` and wait for at least the data has been written or we can say `vista#executive#nvim_lsp#SetData` has been call.
3. But **`g:vista_executive_nvim_lsp_fetching` is not reset to `true` in `s:Run` before `call s:RunAsync()`**, so from second time we call `s:Run`, it will not be blocked to wait for `s:RunAsync` finish, so the `s:data` in `nvim_lsp.vim` will be stale.

**If you agree with the potential bug I found, I can send a pr to fix it**.  Looking forward to your response.  Thank you.

### PR DESCRIPTION
**Pull request by lizhemingi**
Resolve #438

### CODE REVIEW

**Comment by lizhemingi** [comment]
This fix problem in my local environment. 

Please take a look. @liuchengxu

File: autoload/vista/executive/nvim_lsp.vim [hunk 1]
```diff
@@ -55,9 +56,9 @@ function! s:RunAsync() abort
 
         if err then print(tostring(err)) return end
         if not result then return end
-        vim.g.vista_executive_nvim_lsp_fetching = false
         data = vim.fn['vista#renderer#LSPPreprocess'](result)
         vim.fn['vista#executive#nvim_lsp#SetData'](data)
+        vim.g.vista_executive_nvim_lsp_fetching = false
```
Comment (liuchengxu) [inline autoload/vista/executive/nvim_lsp.vim:61]: Is this change really necessary?

Comment (lizhemingi) [reply autoload/vista/executive/nvim_lsp.vim:61]: If not, there still a chance that `Run` stop fetching but `vista#executive#nvim_lsp#SetData` is not call and `s:data` is not overwrite by new data ?

Comment (liuchengxu) [reply autoload/vista/executive/nvim_lsp.vim:61]: Right, `s:Run` needs to block until `s:data` is updated.

**Comment by liuchengxu** [comment]
This will probably fix CI

```diff
diff --git a/.github/workflows/ci.yml b/.github/workflows/ci.yml
index 03d6f80..46f98ba 100644
--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -13,7 +13,7 @@ jobs:
     - name: Checkout
       uses: actions/checkout@master
     - name: Run vint with reviewdog
-      uses: reviewdog/action-vint@v1.0.1
+      uses: reviewdog/action-vint@v1
       with:
         github_token: ${{ secrets.github_token }}
         reporter: github-pr-review
```

**Comment by lizhemingi** [comment]
> This will probably fix CI
> 
> ```diff
> diff --git a/.github/workflows/ci.yml b/.github/workflows/ci.yml
> index 03d6f80..46f98ba 100644
> --- a/.github/workflows/ci.yml
> +++ b/.github/workflows/ci.yml
> @@ -13,7 +13,7 @@ jobs:
>      - name: Checkout
>        uses: actions/checkout@master
>      - name: Run vint with reviewdog
> -      uses: reviewdog/action-vint@v1.0.1
> +      uses: reviewdog/action-vint@v1
>        with:
>          github_token: ${{ secrets.github_token }}
>          reporter: github-pr-review
> ```
Seems a newer version of `reviewdog/action-vint` also fix the problem. Should I file another pr or you will have some other way to deal with it? @liuchengxu

**Comment by lizhemingi** [comment]
I have added a commit int this pr to change "action-vint" to v1, can you have another try on CI ? @liuchengxu
